homeassistant/components/fritzbox_callmonitor/sensor.py

```diff
@@ -208,6 +209,7 @@ def _parse(self, line):
         df_out = "%Y-%m-%dT%H:%M:%S"
         isotime = datetime.datetime.strptime(line[0], df_in).strftime(df_out)
         if line[1] == "RING":
+            self.callstack[line[2]] = ["RING"]
```

Comment:
This integration needs to extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices